src/overlays/actors/ovl_En_Egol/z_en_egol.c

```diff
@@ -119,99 +320,1319 @@ const ActorInit En_Egol_InitVars = {
+        this->laserCollider.dim.quad[0].z = this->laserCollider.dim.quad[2].z + laserVec.z;
+        Collider_SetQuadVertices(&this->laserCollider, &this->laserCollider.dim.quad[0],
+                                 &this->laserCollider.dim.quad[1], &this->laserCollider.dim.quad[2],
+                                 &this->laserCollider.dim.quad[3]);
+        this->laserState = EYEGORE_LASER_ON;
+    }
+    if ((limbIndex == EYEGORE_LIMB_LASER_ATTACH) && (this->laserState == EYEGORE_LASER_START)) {
+        this->chargingLaser = true;
+        this->laserState = EYEGORE_LASER_CHARGING;
+        this->laserScaleTarget.x = 0.03f;
+        this->laserScaleTarget.y = 0.03f;
+    }
+    if (limbIndex == EYEGORE_LIMB_LEFT_FOOT) {
+        Matrix_MultVec3f(&footOffset, &this->leftFootPos);
+    }
+    if (limbIndex == EYEGORE_LIMB_RIGHT_FOOT) {
+        Matrix_MultVec3f(&footOffset, &this->rightFootPos);
+    }
+    if ((limbIndex == EYEGORE_LIMB_HEAD) || (limbIndex == EYEGORE_LIMB_LEFT_SHOULDER) ||
+        (limbIndex == EYEGORE_LIMB_LEFT_ARM) || (limbIndex == EYEGORE_LIMB_LEFT_HAND) ||
+        (limbIndex == EYEGORE_LIMB_RIGHT_SHOULDER) || (limbIndex == EYEGORE_LIMB_RIGHT_ARM) ||
+        (limbIndex == EYEGORE_LIMB_RIGHT_HAND) || (limbIndex == EYEGORE_LIMB_UPPER_EYELID) ||
+        (limbIndex == EYEGORE_LIMB_LOWER_EYELID) || (limbIndex == EYEGORE_LIMB_HIPS) ||
+        (limbIndex == EYEGORE_LIMB_LEFT_SHIN) || (limbIndex == EYEGORE_LIMB_RIGHT_SHIN)) {
+        Matrix_MultZero(&this->limbPos[this->limbPosIndex]);
+        this->limbPosIndex++;
+        if (this->limbPosIndex >= ARRAY_COUNT(this->limbPos)) {
+            this->limbPosIndex = 0;
+        }
+    }
+    Collider_UpdateSpheres(limbIndex, &this->bodyCollider);
+    Collider_UpdateSpheres(limbIndex, &this->eyeCollider);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Egol/func_80A7EA88.s")
+void EnEgol_Draw(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    EnEgol* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Egol/func_80A7EAD8.s")
+    OPEN_DISPS(play->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Egol/func_80A7EB14.s")
+    func_8012C28C(play->state.gfxCtx);
+    if (this->actionState != EYEGORE_ACTION_DEAD) {
+        AnimatedMat_Draw(play, Lib_SegmentedToVirtual(gEyegoreEyeLaserTexAnim));
+        SkelAnime_DrawFlexOpa(play, this->skelAnime.skeleton, this->skelAnime.jointTable, this->skelAnime.dListCount,
+                              EnEgol_OverrideLimbDraw, EnEgol_PostLimbDraw, &this->actor);
+        POLY_OPA_DISP = func_801660B8(play, POLY_OPA_DISP);
+    }
+    if (this->dmgEffectTimer != 0) {
+        f32 effectAlpha = 0.05f * this->dmgEffectTimer;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Egol/func_80A7EB54.s")
+        Actor_DrawDamageEffects(play, &this->actor, this->limbPos, ARRAY_COUNT(this->limbPos), 0.8f, 0.8f, effectAlpha,
```

Comment:
```suggestion
        Actor_DrawDamageEffects(play, &this->actor, this->limbPos, ARRAY_COUNT(this->limbPos), 0.8f, 0.8f, drawDmgEffAlpha,
```
I've made an effort to standardize the names related to this function, would look like this. Could also do `drawDmgEffectTimer`